Configure to promote all resource attributes except ignore ones using 'otlp.ignore_resource_attributes' (#16426)
prometheus/prometheus: docs/configuration/configuration.md

# Settings related to the OTLP receiver feature.
# See https://prometheus.io/docs/guides/opentelemetry/ for best practices.
otlp:
  # Promote specific list of resource attributes to labels.
  # It cannot be configured simultaneously with 'ignore_resource_attributes'.
  [ promote_resource_attributes: [<string>, ...] | default = [ ] ]
  # Promoting all resource attributes to labels, except the ignore ones. 

>   # In case you want to promote all resource attributes, you can set 'ignore_resource_attributes' to an empty list.

Reviewer: Nit: I would call out setting `ignore_resource_attributes` to `[ ]` can cause up to 100+ new labels to be added.

Author: Correct and that is why we have two ways of configuring resoruce attributes.
- In case you want to only promote a list of specific attributes, you will use `promote_resource_attributes`
- If you want to promote all your attributes you will use `ignore_resource_attributes: []` or if you want to promote all your attribute except a list of specific ones you can use  `ignore_resource_attributes`

Thanks :)

>   # Promoting all resource attributes to labels, except the ignore ones. 
>   # In case you want to promote all resource attributes, you can set 'ignore_resource_attributes' to an empty list.
>   # It cannot be configured simultaneously with 'promote_resource_attributes'.

Reviewer 2: ```suggestion
  # Promoting all resource attributes to labels, except the ignored ones. 
  # 'ignore_resource_attributes' can be used to promote all resource attributes if set to an empty list:
  #
  #  otlp:
  #    ignore_resource_attributes: []
  #
  # Be aware that changes in attributes received by the OTLP endpoint may result in time series churn and lead to high memory usage by the Prometheus server.
  # It cannot be configured simultaneously with 'promote_resource_attributes'.
```

Let's make sure users understand the trade-offs of promoting labels without safe guards 😬

# Settings related to the OTLP receiver feature.
# See https://prometheus.io/docs/guides/opentelemetry/ for best practices.
otlp:
  # Promote specific list of resource attributes to labels.
  # It cannot be configured simultaneously with 'promote_all_resource_attributes: true'.
  [ promote_resource_attributes: [<string>, ...] | default = [ ] ]

>   # Promoting all resource attributes to labels.

Reviewer 3: ```suggestion
  # Promoting all resource attributes to labels, except for the ones configured with ignore_resource_attributes.
```

  # Be aware that changes in attributes received by the OTLP endpoint may result in time series churn and lead to high memory usage by the Prometheus server.
  # It cannot be set to 'true' simultaneously with 'promote_resource_attributes'.
  [ promote_all_resource_attributes: <boolean> | default = false ]

>   # Promoting all resource attributes to labels, except the ignored ones. 

Reviewer 3: ```suggestion
  # Which resource attributes to ignore when promote_all_resource_attributes is true.
```

>   # Which resource attributes to ignore when 'promote_all_resource_attributes' is true.
>   [ ignore_resource_attributes: [<string>, ...] | default = [] ]

Reviewer 2: ```suggestion
  # Which resource attributes to ignore, can only be set when 'promote_all_resource_attributes' is true.
  [ ignore_resource_attributes: [<string>, ...] | default = [] ]
```

Could we make it clearer that ignore can only be used together with promote all? :)
